server/src/routes/pollRoute.ts

```diff
@@ -62,7 +62,7 @@ pollRouter.get("/students", async (req, res) => {
 pollRouter.get("/status", async (req, res) => {
   const { pollId } = req.query;
   try {
-    const result = await getPollStatus(pollId);
+    const result = await getPollStatus(pollId as string | undefined);
```

Comment:
A type assertion does not ensure that `pollId` will be a string. I would add a type guard to ensure that `pollId` is a string

Reply:
Well why make a call to an external service (i.e. redis) when we can perform a simple check beforehand

Reply:
> Worse case it will return "[object Object]" which would be an invalid id anyhow

Once it returns "[object Object]" your approach would go ahead and call `getPollStatus` which requires calling Redis. On the other hand, if you assert that it's a string using `typeof pollId == 'string'`, you immediately know it's not a string and avoid the overhead of a Redis call that's guaranteed to fail.

Reply:
Understood, will fix. Thanks for explaining.

Reply:
Fixed in 5d8bbaf